### DISCUSSION TITLE
Altium support

### DISCUSSION THREAD

**Discussion by vetaldudnik** [Part-DB/Part-DB-server#501] [Ideas]
Are there any plans to integrate with Altium dblib?
To do this, Altium needs direct access to the sql database...

**Comment by jbtronics** [comment]
I dont really know how flexible Altium is with the desired database structure. Maybe you can already do it today, and reuse the KiCad fields for Altium footprints and symbols.

In general I am not quite the big fan of directly accessing the database. It circumvents Part-DB user and permission system and every change to Part-DB database schema, could potentially break the altium connection. The latter can maybe resolved by creating a sql view for altium, which structure is fixed. But the support of the database abstraction layer Part-DB uses for views is not really existing (I think you would have to write all sql migrations relating to the view by hand).

**Comment by vetaldudnik** [comment]
Unfortunately, Altium only works using a direct connection to the sql database.
In terms of database structure flexibility - any database field interpretation can be  configured by Altium.
We use a similar system for electronic components, and there interaction with Altium is done through database views.

**Comment by electronics4fun** [comment]
We have PartDb running under mySQL. In the database i defined some sql-views, which are accessed by Altium. This is as a read only system (view from Altium) working fine. If you are interested i can post the sql-views, but i am pretty sure that they can be improved.

My big hope is, that Altium is coming up with a API-Support, like KiCad, in the future.

**Reply by vetaldudnik** [reply]
Sure, I'm interested

**Reply by VirusSE** [reply]
We were currently evaluating Part-db in our company and using Altium too. So maybe it is interesting for a greater amount of people. 
In the strategy of Altium is nothing mentioned to provide something API-like on their side. On the other hand they are currently integrating several part providers. When they don't use provider specific APIs, maybe this API-Support on Altium side come real in the next month. It would make sense to have an API, so that "lightweight" adapters could be used for any other part providers.

**Reply by jbtronics** [reply]
> When they don't use provider specific APIs, maybe this API-Support on Altium side come real in the next month. It would make sense to have an API, so that "lightweight" adapters could be used for any other part providers.

Yeah that would be useful, but we will see what Altium will decide.
In general such an API would be better than direct DB access, as it would allow for proper authentication and is more stable. And its much easier to use HTTPs through non-secure networks than to tunnel a direct MySQL access.

**Comment by electronics4fun** [comment]
I updated and renamed the repository for connecting Altium with PartDb.
Feel free to test it and let me know if there are any issues.
[PartDb-Altium-Connector](https://github.com/electronics4fun/PartDb-Altium-Connector)

**Comment by electronics4fun** [comment]
If you want to support the work here, you can promote the following requests in the altium-forum to give them more attention: 
https://forum.live.altium.com/#/posts/258311
https://forum.live.altium.com/#/posts/258269

**Comment by electronics4fun** [comment]
Hi,

i added a feature for the use of PartDb with KiCad and Altium at the same time.
If you like, test it and let me know if it's working properly.
See [PartDb-Altium-Connector](https://github.com/electronics4fun/PartDb-Altium-Connector)
